dnf-docker-test/features/steps/repo_utils.py

```diff
@@ -0,0 +1,178 @@
+    elem.appendChild(cnt)
+    elem_update.appendChild(elem)
+    # severity
+    elem = doc.createElement("severity")
+    cnt = doc.createTextNode(update.get('Severity', 'Low'))
+    elem.appendChild(cnt)
+    elem_update.appendChild(elem)
+    # issued
+    elem = doc.createElement("issued")
+    elem.setAttribute("date", update.get('Issued', '2017-01-01 00:00:01'))
+    elem_update.appendChild(elem)
+    # updated
+    elem = doc.createElement("updated")
+    elem.setAttribute("date", update.get('Updated', '2017-01-01 00:00:01'))
+    elem_update.appendChild(elem)
+    # rights
+    elem = doc.createElement("rights")
+    cnt = doc.createTextNode(update.get('Rights', 'nobody'))
+    elem.appendChild(cnt)
+    elem_update.appendChild(elem)
+    # summary
+    elem = doc.createElement("summary")
+    cnt = doc.createTextNode(update.get('Summary', 'Default summary of %s' % update['Id']))
+    elem.appendChild(cnt)
+    elem_update.appendChild(elem)
+    # description
+    elem = doc.createElement("description")
+    cnt = doc.createTextNode(update.get('Description', 'Default description of %s' % update['Id']))
+    elem.appendChild(cnt)
+    elem_update.appendChild(elem)
+    # solution
+    elem = doc.createElement("solution")
+    cnt = doc.createTextNode(update.get('Solution', 'Default solution of %s' % update['Id']))
+    elem.appendChild(cnt)
+    elem_update.appendChild(elem)
+    # references
+    references = doc.createElement("references")
+    cnt = doc.createElement("reference")
+    cnt.setAttribute("href", "www.path.to/nowhere")
+    cnt.setAttribute("type", "self")
+    cnt.setAttribute("title", update['Id'])
+    references.appendChild(cnt)
+    if 'Reference' in update:
+        for ref in update['Reference']:
+            cnt = doc.createElement("reference")
+            cnt.setAttribute("href", "www.path.to/%s" % ref)
+            cnt.setAttribute("title", ref)
+            if ref.startswith('BZ'):
+                cnt.setAttribute("id", ref[2:])
+                cnt.setAttribute("type", 'bugzilla')
+            if ref.startswith('CVE'):
+                cnt.setAttribute("id", ref)
+                cnt.setAttribute("type", 'cve')
+            references.appendChild(cnt)
+    elem_update.appendChild(references)
+    # pkglist
+    pkglist = doc.createElement("pkglist")
+    elem_update.appendChild(pkglist)
+    # collection
+    collection = doc.createElement("collection")
+    collection.setAttribute("short", update.get('Collection', 'Default collection'))
+    pkglist.appendChild(collection)
+    # collection <name>
+    elem = doc.createElement("name")
+    cnt = doc.createTextNode(update.get('Collection', 'Default collection'))
+    elem.appendChild(cnt)
+    collection.appendChild(elem)
+    # now for every package I need to prepare this based on the details obtained from primary.xml
+    # <package name="PKG_NAME" version="3.0.33" release="3.29.el5_5.1" epoch="0" arch="i386" src="samba-3.0.33-3.29.el5_5.1.src.rpm">
```

Comment:
All issues should be fixed now. I have also added a simple test verifying the new functionality.